Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -300,10 +306,18 @@ import BraintreeDataCollector
         
         authenticationSession?.presentationContextProvider = self
         returnedToAppAfterPermissionAlert = false
+        apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationStarted)
         isAuthenticationSessionStarted = authenticationSession?.start() ?? false
 
         let authenticationSessionStatus = isAuthenticationSessionStarted ? "succeeded" : "failed"
-        apiClient.sendAnalyticsEvent("ios.\(paymentType.stringValue).authsession.start.\(authenticationSessionStatus)")
+        if authenticationSessionStatus == "failed" {
+            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.browserPresentationFailed)
+            apiClient.sendAnalyticsEvent(BTPayPalAnalytics.tokenizeFailed)
+            completion(nil, BTPayPalError.webSessionError)
+        }
```

Comment:
```suggestion
            completion(nil, BTPayPalError.webSessionError)
            return
        }
```

Reply:
I don't think we want to return there because from what I observed, it launches into the webSession after this.

Reply:
oops, good catch